### PR TITLE
CMake: Generate and install CMake config files

### PR DESCRIPTION
**Pull request by sfuller**
This change should make importing glad into consumer CMake projects a lot easier, by being able to import glad using `find_package(glad CONFIG REQUIRED)` and reference include directories and link libraries with `glad::glad`

### CODE REVIEW

**Comment by Dav1dde** [comment]
Thanks for the PR, sorry it took me forever to come back to you.

The main problem is my cmake is very limited. Maybe just walk me through it shortly and how I can verify that it works (so nothing in the future breaks it again).

Also is it a breaking change? It appears to only potentially break the install target?

File: CMakeLists.txt [hunk 1]
```diff
@@ -100,10 +106,51 @@ endif()
 
 # Install
 if(GLAD_INSTALL)
+
+  set(config_install_dir "lib/cmake/glad")
+  set(version_config "${CMAKE_CURRENT_BINARY_DIR}/gladConfigVersion.cmake")
+  set(project_config "${CMAKE_CURRENT_BINARY_DIR}/gladConfig.cmake")
+  set(targets_export_name "gladTargets")
+  set(namespace "glad::")
```
Comment (sfuller) [inline CMakeLists.txt]: I've added a bunch of variables here to make editing paths and other install related properties easier.

File: CMakeLists.txt [hunk 2]
```diff
@@ -100,10 +106,51 @@ endif()
...
+    LIBRARY DESTINATION "lib"
+    ARCHIVE DESTINATION "lib"
+    RUNTIME DESTINATION "bin"
+    INCLUDES DESTINATION "include")
+
   if(GLAD_INCLUDE_DIRS)
-    install(DIRECTORY ${GLAD_INCLUDE_DIRS} DESTINATION ${CMAKE_INSTALL_PREFIX})
+    install(
+      DIRECTORY ${GLAD_INCLUDE_DIRS}
+      DESTINATION ${CMAKE_INSTALL_PREFIX})
```
Comment (sfuller) [inline CMakeLists.txt]: Feel free to put these parameters back into one line.

File: Config.cmake.in [hunk 3]
```diff
@@ -0,0 +1,4 @@
+@PACKAGE_INIT@
+
+include("${CMAKE_CURRENT_LIST_DIR}/@targets_export_name@.cmake")
+check_required_components(glad)
```
Comment (sfuller) [inline Config.cmake.in]: This file is the template for the glad's CMake package configuarion file. It is used to generate gladConfig.cmake, which is installed to <prefix>/lib/cmake/glad. Having this file installed (along with the other 'targets' and 'versions' cmake files) allows the package to be used with CMake's newer package configuration system. No more writing 'FindXXX.cmake' files by hand! 
Using the newer package configuration system, glad can be included in a project just by installing it to a prefix and including the following lines of CMake:
```
find_package(glad CONFIG REQUIRED)
...
target_link_libraries(target glad::glad)
# Note: include directories are also imported by the above line, just by using the glad target.
```
(Note that "glad::" is the value namespace variable defined in the CMakeLists.txt file)

**Comment by sfuller** [comment]
> Also is it a breaking change? It appears to only potentially break the install target?

Yes, good catch. I changed the name of the install target to gladTargets to match CMake convention, feel free to change this back to glad-targets. You should be able to set that back by changing `set(targets_export_name "gladTargets")` to `set(targets_export_name "glad-targets")`

Let me know if you have any more questions :)

**Comment by Dav1dde** [comment]
Thanks, for your work!

I pushed everything in one commit (with the merge it got a bit hard to track). I played around with it locally, it seems to work but would be good if could test it again.

**Pull request closed without merging**